platform/duration.cpp

```diff
@@ -0,0 +1,104 @@
+#include "duration.hpp"
+
+namespace platform
+{
+namespace
+{
+using std::chrono::duration_cast, std::chrono::seconds, std::chrono::minutes, std::chrono::hours, std::chrono::days;
+
+long SecondsToUnits(seconds duration, Duration::Units unit)
+{
+  switch (unit)
+  {
+    case Duration::Units::Days: return duration_cast<days>(duration).count();
+    case Duration::Units::Hours: return duration_cast<hours>(duration).count();
+    case Duration::Units::Minutes: return duration_cast<minutes>(duration).count();
+    default: UNREACHABLE();
+  }
+}
+
+seconds UnitsToSeconds(long value, Duration::Units unit)
+{
+  switch (unit)
+  {
+    case Duration::Units::Days: return days(value);
+    case Duration::Units::Hours: return hours(value);
+    case Duration::Units::Minutes: return minutes(value);
+    default: UNREACHABLE();
+  }
+}
+
+std::string GetUnitSeparator(Locale const &locale)
+{
+  static const std::unordered_set<std::string> kEmptyNumberUnitSeparatorLocales =
+  {
+    "en", "de", "fr", "he", "fa", "ja", "ko", "mr", "th", "tr", "vi", "zh"
+  };
+  return kEmptyNumberUnitSeparatorLocales.count(locale.m_language) ? "" : kNarrowNonBreakingSpace;
```

Comment:
kNoSpace = ""?

Will returning string_view instead of string work?

Reply:
replaced with string_view